### ISSUE TITLE
Zeroed fields by default for proto3?

### ISSUE THREAD

**Issue by acetylen** [eigenein/protobuf#84]
I noticed a part of what was brought up in #63 wasn't really discussed, namely the fact that proto3 doesn't have the concept of "nullable". Any field not specified is left at whatever is considered to be "zero" for that particular type, and left out of the message. This clashes with how the library currently behaves, where initializing a message without specifying every field results in an error.

This would remove the need for `Optional` hints and the `default` keyword (although that would still be a nice-to-have) in definitions, while of course sacrificing compatibility with proto2.

**Comment by eigenein** [comment]
Good point. Though at the moment I tend to think it's better to leave it like that because of multiple reasons:

- Proto2 and Proto3 are just schema syntaxes, the wire format stays the same. Unsure if I want Python code to be an exact adaptation of the `.proto` syntaxes
- If I implement the defaults, this would break the usual constructor/initializer behaviour. I sort of like that not specifying each non-default field results in an error, just like with a normal `dataclass`. So I don't have to care whether I instantiating a `@message`-decorated type
- Addition to the previous point: not seeing a way not to break the current `@message` interface. The change would suddenly make all required fields in exiting code non-required. I could bump the major version of course, but still I have the points above

I may be mistaken of course, so open for objections or suggestions to that
